17. As a user I want my lists protected by google and firebase authentication. I want the log in/ sign up to be separated from the rest of the app for clarity and protection/privacy.

StefieCaff opened the pull request:
## Description

This code introduces our branding and guides the user to use their google account to use our app. 

For concise and clear functionality  the login/ sign up button is shared and labeled with google authentication.  

A not found page was also added. In the event the user tries to navigate to a non existing page, they will be navigated here. A message will alert them and a link will allow them to navigate back to the Login page, which in turn will navigate them home if they are authenticated, or allow them to re-authenticate.

I would like to add other forms of authentication in the event a user chooses not to use google.

## Related Issue
closes #41 

## Acceptance Criteria

- [x] The signup page resembles the figma design and basic theming
- [x] The layout should be responsive based on screen sizes
- [x] Only the sign up page is navigable without permissions. 

## Type of Changes

`accessibility` or `enhancement` `security`

## Updates

### Before

![issue-17-before-2](https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/108297341/79b8a52a-a6b9-4100-932a-28c1835402c4)

### After

![issue-17-after](https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/108297341/7abc3ccc-e78b-4a2c-acd1-aef4fdde83ca)

## Testing Steps / QA Criteria

* git pull from main
* git checkout sc-issue-17
* npm start
* sign out if you are already signed in
* copy and past this into the browser address bar to attempt to access the home page: http://localhost:3000/
* copy and past this into the browser address bar to attempt to access the list page: http://localhost:3000/list
* copy and past this into the browser address bar to access the not found page: http://localhost:3000/kawkaw
* sign up/ log in to the app. Navigate to the home and list page. Sign out of the app. Repeat as many times as desired.
*  inspect the page for screen size compatibility down to 280 px ( Galaxy Fold)

StefieCaff committed:
Create components for signed up Gate

PublicRoute and PrivateRoute components can wrap components in routing. They will conditionally render the component or navigate to appropriate view depending on auth status from useAuth() hook in utils

StefieCaff committed:
implement public and private routes

add landing page and err page
add spinner

StefieCaff committed:
update loading state

add spinner when loading list data

StefieCaff committed:
styling and renaming

add spinner component to list.jsx
change landing to login

StefieCaff committed:
Style login page

add icons
add graph background

StefieCaff committed:
responsive login page

Co-Authored-By: eonflower <98231995+eonflower@users.noreply.github.com>

StefieCaff committed:
lighten background image

StefieCaff committed:
change login to verify

if the user is signing up  or logging in this is more clear for both options

StefieCaff committed:
update file path

3campos reviewed (changes requested):
Good job on this! I ran the code locally. Navigating the web pages worked as expected when signed in and not signed in. Nice work on styling the login page.

Two changes that I think should be made (regarding the error page) before merging are:

1. Add a visible link (or alternative) to the error page so that the user can click and return to the home page. You can make the call regarding how to approach this. Some options that come to my mind are: format a piece of the text as a simple visible hyperlink; make a button; or have the page automatically redirect after 2 to 3 seconds.
2. Since this error page will only appear when a user is already signed in, it sounds better if the last sentence of the page ("Try logging back in.") is replaced with something such as: "Click here to be redirected to the home page". If you're using an automatic redirect, then use something like: "Don't worry, we're sending in an extraction team to bring you back home."

One thing I noticed that is not a blocker is that the screen briefly flashes entirely in a dark color before the loading spinner appears. I took a screen recording of this happening. It's a very brief flash and we can create another issue ("bug fix"), if you think it negatively impacts the user experience. Again, this is not a blocker.


https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/105468537/07f0f5b6-a6fd-4bb1-aa3c-8514044f7366

StefieCaff committed:
update background

add color for contrast

StefieCaff committed:
accessibility

add alert and navigation to sign out so url will refresh
add aria labels and alt tags where needed

StefieCaff committed:
Style Err.jsx

StefieCaff committed:
responsive down to 280px

3campos reviewed (approved):
The darker graph paper lines contrast better with our light background and you made good changes to the error page. Merge when you're ready!

nick-zanetti commented on src/views/Err.jsx:
```diff
@@ -0,0 +1,37 @@
+					<div className="inline-block border-t-1 self-stretch border-loginLine/50"></div>
+					<div className="px-4">
+						<h3>
+							Please navigate back to the{' '}
```
Maybe this could just say "Click here to navigate back to your lists"?

StefieCaff replied:
yeah, that sounds better for sure, thanks!

nick-zanetti reviewed (approved):
This looks good to me! One thought: when users click the "verify with google" button, is there a way they can choose a specific google account to use? Does this automatically use a google account if you are logged in to google with it on the browser?

StefieCaff committed:
style Err.jsx, refactor signInButton

reword user message
add functionality for FireFox --- getRedirectResult, signInWithPopup

StefieCaff committed:
adjust background graph color

lighten background on login and err views
